app/src/main/java/net/sourceforge/opencamera/SyncSettingsContainer.java

```diff
@@ -0,0 +1,136 @@
+     */
+    public SyncSettingsContainer(MainActivity mainActivity) {
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(mainActivity);
+        Preview preview = mainActivity.getPreview();
+        CameraController cameraController = preview.getCameraController();
+
+        syncISO = sharedPreferences.getBoolean(PreferenceKeys.SyncIsoPreferenceKey, false);
+        syncWb = sharedPreferences.getBoolean(PreferenceKeys.SyncWbPreferenceKey, false);
+        syncFlash = sharedPreferences.getBoolean(PreferenceKeys.SyncFlashPreferenceKey, false);
+        syncFormat = sharedPreferences.getBoolean(PreferenceKeys.SyncFormatPreferenceKey, false);
+
+        isVideo = preview.isVideo();
+        exposure = cameraController.captureResultExposureTime();
+        iso = cameraController.captureResultIso();
+        wbTemperature = cameraController.captureResultWhiteBalanceTemperature();
+        wbMode = cameraController.getWhiteBalance();
+        flash = cameraController.getFlashValue();
+        format = isVideo ?
+                sharedPreferences.getString(PreferenceKeys.VideoFormatPreferenceKey, "preference_video_output_format_default") :
+                sharedPreferences.getString(PreferenceKeys.ImageFormatPreferenceKey, "preference_image_format_jpeg");
+    }
+
+    /**
+     * The container saves the provided values.
+     *
+     * @param syncISO whether to sync ISO or not.
+     * @param syncWb whether to sync white balance or not.
+     * @param syncFlash whether to sync flash mode or not.
+     * @param syncFormat whether to sync video/image format or not.
+     * @param isVideo capture mode (true if video, false if photo).
+     * @param exposure exposure time.
+     * @param iso ISO value.
+     * @param wbTemperature white balance temperature.
+     * @param wbMode white balance mode.
+     * @param flash flash mode.
+     * @param format video/image format (must match the chosen capture mode).
+     */
+    public SyncSettingsContainer(boolean syncISO, boolean syncWb, boolean syncFlash, boolean syncFormat,
+                                 boolean isVideo, long exposure, int iso, int wbTemperature, String wbMode, String flash, String format) {
+        this.syncISO = syncISO;
+        this.syncWb = syncWb;
+        this.syncFlash = syncFlash;
+        this.syncFormat = syncFormat;
+
+        this.isVideo = isVideo;
+        this.exposure = exposure;
+        this.iso = iso;
+        this.wbTemperature = wbTemperature;
+        this.wbMode = wbMode;
+        this.flash = flash;
+        this.format = format;
+    }
+
+    /**
+     * Constructs a string with this settings serialized using {@link ByteArrayOutputStream} and
+     * {@link ObjectOutputStream}.
+     * <p>
+     * The string is only constructed once and cached, the cached value is returned from all
+     * subsequent invocations.
+     *
+     * @return a constructed string.
+     */
+    public String asString() {
```

Comment:
Why not toString?

Reply:
Docs for `toString` say that "The result should be a concise but informative representation that is easy for a person to read", but this method returns a serialized version of the object, which is not really intended  to be read by a person.

Reply:
okay, you say that it serializes object - so it should be mentioned in the name -> serializeToString()